library/src/test/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionBindingTest.java

```diff
@@ -33,9 +45,49 @@
     @Before
     public void setup() {
         MockitoAnnotations.initMocks(this);
-        sectionAdapter = new SectionedRecyclerViewAdapter();
-        sectionAdapter.addSection(dummySection);
-        sectionAdapter.addSection(section);
+        sectionAdapter = new AdapterImpl();
+        sectionAdapter.addSection(dummySection); // View types 0-5, items 0-9
+        sectionAdapter.addSection(section); // View types 6-11, items 10-19
+    }
+
+    @Test
+    public void onCreateViewHolder_withItemResourceId_calls_getItemViewHolder() {
+        // When
+        sectionAdapter.onCreateViewHolder(null, 6 + SectionedRecyclerViewAdapter.VIEW_TYPE_ITEM_LOADED);
+
+        // Then
+        verify(section, never()).getItemView(any(ViewGroup.class));
+        verify(section, times(1)).getItemViewHolder(argThat(hasTag(-1)));
+    }
+
+    @Test
+    public void onCreateViewHolder_withItemViewProvided_calls_getItemView() {
+        // Given
+        Section providingSection = spy(new SectionImpl(new SectionParameters.Builder().itemViewWillBeProvided()));
+        sectionAdapter.addSection(providingSection); // Third section, view types 12-17
+        doReturn(dummyViewWithTag(42)).when(providingSection).getItemView(null);
+
+        // When
+        sectionAdapter.onCreateViewHolder(null, 12 + SectionedRecyclerViewAdapter.VIEW_TYPE_ITEM_LOADED);
+
+        // Then
+        verify(providingSection, times(1)).getItemView(null);
+        verify(providingSection, times(1)).getItemViewHolder(argThat(hasTag(42)));
```

Comment:
I liked the strategy used here to create the ArgumentMatcher instead of using an ArgumentCaptor 👍